src/SystemReport.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Krokedil\Support;
+
+use Automattic\WooCommerce\Internal\Admin\Logging\Settings;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class SystemReport
+ *
+ * @package Krokedil\Support
+ */
+class SystemReport {
+
+	public const REGISTRY_OPTION = 'krokedil_support_registry';
+
+	/**
+	 * The gateway ID.
+	 *
+	 * @var string
+	 */
+	private $id;
+
+	/**
+	 * The gateway name (or title).
+	 *
+	 * @var string
+	 */
+	private $name;
+
+	/**
+	 * SystemReport constructor.
+	 *
+	 * @param string $id The gateway ID.
+	 * @param string $name The gateway name (or title).
+	 */
+	public function __construct( $id, $name ) {
+		$this->id   = $id;
+		$this->name = $name;
+		$this->register( $id );
+
+		add_action( 'woocommerce_system_status_report', array( $this, 'add_status_page_box' ) );
+		add_action( 'woocommerce_cleanup_logs', array( __CLASS__, 'remove_old_entries' ) );
+	}
+
+	/**
+	 * Register gateway gateway system report.
+	 *
+	 * @param string $id The gateway ID.
+	 */
+	private function register( $id ) {
+		$registry   = get_option( self::REGISTRY_OPTION, array() );
+		$registry[] = $id;
+
+		update_option( self::REGISTRY_OPTION, array_unique( $registry ) );
+	}
```

Comment:
I would suggest to check if it already exists in the array we don't add it or update the option. That saves a database execution for updating the option on every pageload.